src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -41,7 +42,7 @@ namespace Microsoft.ML.Runtime.PCA
     /// <remarks>
     /// This PCA can be made into Kernel PCA by using Random Fourier Features transform
     /// </remarks>
-    public sealed class RandomizedPcaTrainer : TrainerBase<PcaPredictor>
+    public sealed class RandomizedPcaTrainer : TrainerEstimatorBase<BinaryPredictionTransformer<PcaPredictor>, PcaPredictor>
```

Comment:
>BinaryPredictionTransformer [](start = 68, length = 27)

So, why is this a binary prediction transformer? Prior to your change it was not a binary predictor, it was part of the anomaly detection task.

Reply:
Good point, thanks. 

---
In reply to: [220048006](https://github.com/dotnet/machinelearning/pull/996#discussion_r220048006) [](ancestors = 220048006)